### DISCUSSION TITLE
deploying from local to remote server

### DISCUSSION THREAD

**Discussion by Anaxarchos** [pimcore/pimcore#13133] [Q&A / Support] [answered]
Hi to all,
I built a small website with pimcore on my local machine and saved it on a git repository.
Now I want to deploy it on a remote server.
When I compose a fresh installation of pimcore with composer, everything is working fine, but I cannot clone my git repository into that application.
When I clone my repository to the server, I have no vendor directory and so I do not know how to setup pimcore. 
In the documentation I did not find a proper way to set this up.
How are you doing this?
Thank you for your anwsers.

**Comment by BlackbitDevs** [comment, marked as answer]
Please have a look at https://github.com/pimcore/pimcore/discussions/13071#discussioncomment-3603868 - there it is described how to do Pimcore updates but almost the same applies for normal deployments:
> You then put these changed files to the live server, execute composer install to install the libraries from composer.lock, run bin/console doctrine:migrations:migrate (or [put this into your composer.json to be run automatically](https://pimcore.com/docs/pimcore/current/Development_Documentation/Development_Tools_and_Details/Migrations.html#page_Run-all-available-migrations-after) - but for post-install-cmd) and perhaps also pimcore:deployment:classes-rebuild which updates database structure according to data object class definition files.

**Reply by Anaxarchos** [reply]
Thank you for your answer.
I did not write a package which I could add to the composer.json file and then pull it in. 
So I still have no idea, how I can put all my classes, template files, javascript and css files to my staging server without using a git pull.

**Reply by BlackbitDevs** [reply]
However you like, you can use `git pull`, FTP, SCP or whatever. The important part is to execute `composer install` and the other commands listed above afterwards.

**Reply by Anaxarchos** [reply]
Thanks, this worked fine.

**Reply by BlackbitDevs** [reply]
Please mark the question as "answered". Thanks.

**Comment by Anaxarchos** [comment]
It seems, that someone else did this already

> Am 19.09.2022 um 08:02 schrieb Blackbit ***@***.***>:
> 
> 
> Please mark the question as "answered". Thanks.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/pimcore/pimcore/discussions/13133#discussioncomment-3677768>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ACIKQQLOVDA5CIUMHTOGLXDV676WPANCNFSM6AAAAAAQKU4UNI>.
> You are receiving this because you authored the thread.
> 

vorneweg.de

Andreas Walter
Woltmershauser Str. 150 
28197 Bremen

Tel: 0170 - 90 28 113
Email: ***@***.***
